### PR TITLE
fix: SSRF in Image/Gallery SVG and Audio postprocessing (GHSA-3xvj-7669-6whx)

### PR DESCRIPTION
**Pull request by abidlabs**
Fixes the read-SSRF reported in [GHSA-3xvj-7669-6whx](https://github.com/gradio-app/gradio/security/advisories/GHSA-3xvj-7669-6whx) (CWE-918).

`gradio.image_utils.extract_svg_content` (the `Image`/`Gallery` SVG postprocess path) fetched user-influenced URLs with a bare `httpx.get(...)` — no private-IP filter, domain allow-list, scheme allow-list, or redirect re-validation. The response body was then inlined into a `data:image/svg+xml,…` URL and returned to the client, exfiltrating internal responses (cloud metadata, internal dashboards, etc.). This contrasts with the sister function `processing_utils.async_ssrf_protected_download`, which already routes through `safehttpx`.

Also adds a semgrep rule to prevent this regressions!

---

> [!NOTE]
> **Medium Risk**
> Changes server-side HTTP behavior for user-controlled URLs (allow-list, redirect handling); incorrect implementation could break legitimate remote media or leave SSRF gaps, but the change is narrowly scoped with tests and static analysis.
> 
> **Overview**
> This PR closes a **server-side request forggery (SSRF)** gap (GHSA-3xvj-7669-6whx) where user-supplied URLs were fetched with plain **`httpx`** and the response could be returned to the client (notably SVG text inlined into `data:image/svg+xml,…`).
> 
> **`processing_utils.async_ssrf_protected_get`** is added as the shared path: requests go through **`safehttpx`** with **`PUBLIC_HOSTNAME_WHITELIST`**, follow redirects with **`urljoin`** (up to 20 hops), and re-validate each target. **`async_ssrf_protected_download`** now uses this helper instead of its own redirect loop.
> 
> Call sites are switched to the protected helper: **`image_utils.extract_svg_content`** (Image/Gallery SVG postprocess) and **`Audio.stream_output`** when the stream path is an HTTP(S) URL.
> 
> A **Semgrep** rule flags bare **`httpx.*`** URL fetches under `gradio/components/`, `image_utils.py`, and `processing_utils.py`. Tests are updated to mock the new helper and cover private-IP rejection and redirect edge cases.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 700e0672484f112b03ace8a2012362c4d806cfb4. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: gradio/image_utils.py [hunk 1]
```diff
@@ -250,8 +250,13 @@ def extract_svg_content(image_file: str | Path) -> str:
     """
     image_file = str(image_file)
     if is_http_url_like(image_file):
-        response = httpx.get(image_file)
-        response.raise_for_status()  # Raise an error for bad status codes
+        # Route through safehttpx to apply SSRF protection (private-IP filter,
+        # domain allow-list, and redirect re-validation) instead of a bare
+        # httpx request, which would otherwise leak internal responses.
```
Comment (abidlabs) [inline gradio/image_utils.py]:
```suggestion
```

File: gradio/components/audio.py [hunk 2]
```diff
@@ -353,7 +353,11 @@ async def stream_output(
                 "extension": ".aac",
             }, output_file
         if client_utils.is_http_url_like(value["path"]):
-            response = httpx.get(value["path"])
+            # Route through safehttpx for SSRF protection rather than a bare
+            # httpx request, which would otherwise leak internal responses.
```
Comment (abidlabs) [inline gradio/components/audio.py]:
```suggestion
```

**Review by hysts — APPROVED** [review]
Thanks! LGTM!
One thing: `async_ssrf_protected_get` is missing the private-request test that `async_ssrf_protected_download` has. Could we mirror `test_private_request_fail` for it?

**Comment by abidlabs** [comment]
Thanks @hysts for the review! Everythign should be good to go now
